core-js/src/definitions.ts

```diff
@@ -0,0 +1,1459 @@
+        throw buildError(`Cannot add built-in ${definition} to this schema (built-ins can only be added at schema construction time)`);
+      }
+    } else {
+      this._directives.set(definition.name, definition);
+    }
+    SchemaElement.prototype['setParent'].call(definition, this);
+    return definition;
+  }
+
+  clone(builtIns?: BuiltIns): Schema {
+    const cloned = new Schema(builtIns ?? this.builtIns);
+    copy(this, cloned);
+    return cloned;
+  }
+}
+
+export class SchemaDefinition extends SchemaElement<Schema, never>  {
+  readonly kind: 'SchemaDefinition' = 'SchemaDefinition';
+  protected readonly _roots: Map<SchemaRoot, ObjectType> = new Map();
+
+  get coordinate(): string {
+    return '';
+  }
+
+  get roots(): ReadonlyMap<SchemaRoot, ObjectType> {
+    return this._roots;
+  }
+
+  root(rootType: SchemaRoot): ObjectType | undefined {
+    return this._roots.get(rootType);
+  }
+
+  setRoot(rootType: SchemaRoot, type: ObjectType): ObjectType;
+  setRoot(rootType: SchemaRoot, name: string, source?: ASTNode): ObjectType;
+  setRoot(rootType: SchemaRoot, nameOrType: ObjectType | string, source?: ASTNode): ObjectType {
+    let toSet: ObjectType;
+    if (typeof nameOrType === 'string') {
+      this.checkUpdate();
+      const obj = this.schema()!.type(nameOrType);
+      if (!obj) {
+        throw new GraphQLError(`Cannot set schema ${rootType} root to unknown type ${nameOrType}`, source);
+      } else if (obj.kind != 'ObjectType') {
+        throw new GraphQLError(`Cannot set schema ${rootType} root to non-object type ${nameOrType} (of type ${obj.kind})`, source);
+      }
+      toSet = obj;
+    } else {
+      this.checkUpdate(nameOrType);
+      toSet = nameOrType;
+    }
+    this._roots.set(rootType, toSet);
+    this.source = source;
+    addReferenceToType(this, toSet);
+    return toSet;
+  }
+
+  protected removeTypeReference(toRemove: NamedType) {
+    for (const [root, type] of this._roots) {
+      if (type == toRemove) {
+        this._roots.delete(root);
+      }
+    }
+  }
+
+  remove(): never[] {
+    if (!this._parent) {
+      return [];
+    }
+    // We don't want to leave the schema without a SchemaDefinition, so we create an empty one. Note that since we essentially
+    // clear this one so we could leave it (one exception is the source which we don't bother cleaning). But it feels
+    // more consistent not to, so that a schemaElement is consistently always detached after a remove()).
+    Schema.prototype['resetSchemaDefinition'].call(this._parent);
+    this._parent = undefined;
+    for (const directive of this._appliedDirectives) {
+      directive.remove();
+    }
+    // There can be no other referencers than the parent schema.
+    return [];
+  }
+
+  toString() {
+    return `schema[${[...this._roots.keys()].join(', ')}]`;
+  }
+}
+
+export class ScalarType extends BaseNamedType<OutputTypeReferencer | InputTypeReferencer> {
+  readonly kind: 'ScalarType' = 'ScalarType';
```

Comment:
I think you should be able to use [`as const`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) here to avoid specifying the string literal twice.

Reply:
Thanks for the tip!